src/Plugin/KeyProvider/EnvironmentVariablesKeyProvider.php

```diff
@@ -65,17 +97,11 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
    * {@inheritdoc}
    */
   public function validateConfigurationForm(array &$form, FormStateInterface $form_state) {
-    $missing_env_variables = [];
-    foreach ($this->getEnvironmentVariables(TRUE) as $variable) {
-      if (!getenv($variable)) {
-        $missing_env_variables[] = $variable;
-      }
+    try {
+      $this->checkRequirements($form_state->getFormObject()->getEntity());
     }
-
-    if (!empty($missing_env_variables)) {
-      $form_state->setError($form, $this->t('The following environment variables are not set: @missing_env_variables.', [
-        '@missing_env_variables' => implode(', ', $missing_env_variables),
-      ]));
+    catch (KeyProviderRequirementsException $exception) {
+      $form_state->setError($form, $exception->getTranslatableMarkupMessage());
```

Comment:
Defensive programming, the `getTranslatableMarkupMessage()` can still be null. At least I would use this simple pattern `$form_state->setError($form, $exception->getTranslatableMarkupMessage() ?? $this->t('Key provider error: @error', ['@error' => $e->getMessage()]))`

Hm, maybe even better. Use ` $this->t('Key provider error: @error', ['@error' => $e->getMessage()])` as default fallback return value in `getTranslatableMarkupMessage()` instead of null. That would simplify usage of this exception.